crates/graphql/src/schema.rs

```diff
@@ -497,6 +497,18 @@ impl Nft {
     pub async fn owner(&self, ctx: &AppContext) -> Option<NftOwner> {
         ctx.nft_owner_loader.load(self.mint_address.clone()).await
     }
+    pub async fn offers(&self, ctx: &AppContext) -> Vec<PublicBid> {
+        ctx.public_bid_loader.load(self.mint_address.clone()).await
+    }
+
+    pub async fn listing(&self, ctx: &AppContext) -> Vec<AuctionHouseListing> {
```

Comment:
This one is interesting because there can only be 1 listing for the auction house but the same nft can be listed on multiple auction houses.

We can do this 2 ways:

1. Make listing take an argument of ah so we can return the single potential listing for the nft.
2. Make this field `listings` and return vec[listings] like you are now. 

I advocate for the 2nd approach and the client filters this list based on the ah scope of the marketplace.

Reply:
The other item is the use of 2 loaders. The two dedicated loaders is good if most clients will query the nft owner in the same requests. Though this is true for our case I still advocate for the auction_house_listing loader to handle the token_account lookup.

The loader should take mint addresses query token_accounts where mint_address = nft.mint_address AND amount = 1. Then inner join on listings with token_account.owner_address = listing.seller AND listing.activated_at IS NOT NUL AND listing.canceled_at IS NUL.

Reply:
```graphql
nfts {
  listings {
       price
          tradeState
           auctionHouse
      }
  }
}
```

```sql
SELECT * LISTINGS INNER JOIN listing.seller = token_accounts.owner 
\ WHERE listing.token_mint = nft.mint_address AND token_account.amount = 1;
```


Discussed in standup this is the query to use.